web/controllers/password_reset_controller.ex

```diff
@@ -10,29 +10,20 @@ defmodule CodeCorps.PasswordResetController do
   2. password & password_confirmation match
   and return email. 422 if pwd do not match or auth token does not exist
   """
-  def reset_password(conn, %{"token" => token, "password" => password, "password_confirmation" => password_confirmation}) do
-    user = conn.assigns.current_user
-    with %AuthToken{value: auth_token} <- Repo.get_by(CodeCorps.AuthToken, %{ value: token, user_id: user.id }),
-      {:ok, _} <- Phoenix.Token.verify(CodeCorps.Endpoint, "user", auth_token, max_age: 1209600) do
-        with %Changeset{valid?: true} <- User.reset_password_changeset(user, 
-                                                                       %{password: password, password_confirmation: password_confirmation}) do
-          conn
-          |> put_status(:created)
-          |> render("show.json", email: user.email)
-        else
-          %Changeset{valid?: false} -> 
-            handle_reset_pswd_result(conn)
-        end
+  def reset_password(conn, %{"token" => reset_token, "password" => _password, "password_confirmation" => _password_confirmation} = params) do
+    with %AuthToken{user: user} <- AuthToken |> Repo.get_by(%{ value: reset_token }) |> Repo.preload(:user),
+         {:ok, _} <- Phoenix.Token.verify(CodeCorps.Endpoint, "user", reset_token, max_age: 1209600),
+         {:ok, updated_user} <- user |> User.reset_password_changeset(params) |> Repo.update,
+         {:ok, auth_token, _claims} = updated_user |> Guardian.encode_and_sign(:token)
+    do
+      conn
+      |> Plug.Conn.assign(:current_user, updated_user)
+      |> put_status(:created)
+      |> render("show.json", token: auth_token, user_id: updated_user.id, email: updated_user.email)
     else
-      nil ->
-        handle_reset_pswd_result(conn)
+      {:error, %Changeset{} = _changeset} -> conn |> put_status(422) |> render(CodeCorps.ErrorView, "422.json")
```

Comment:
Ah now it makes much more sense.  I had assumed in the past that JaResource had handled 422 errors.